src/main/java/de/zalando/aruha/nakadi/util/FeatureToggleService.java

```diff
@@ -4,31 +4,54 @@
 import org.apache.zookeeper.data.Stat;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Value;
 
 public class FeatureToggleService {
 
     private static final Logger LOG = LoggerFactory.getLogger(FeatureToggleService.class);
 
-    @Value("${nakadi.featureToggle.enableAll}")
-    private boolean forceEnableAll;
+    private static final String PREFIX = "/nakadi/feature_toggle/";
 
+    private final boolean forceEnableAll;
     private final ZooKeeperHolder zkHolder;
 
-    public FeatureToggleService(final ZooKeeperHolder zkHolder) {
+    public FeatureToggleService(boolean forceEnableAll, final ZooKeeperHolder zkHolder) {
+        this.forceEnableAll = forceEnableAll;
         this.zkHolder = zkHolder;
     }
 
-    public boolean isFeatureEnabled(final String feature) {
+    public boolean isFeatureEnabled(final Feature feature) {
         if (forceEnableAll) {
-            return true;
+            return feature.getDefault();
         }
         try {
-            final Stat stat = zkHolder.get().checkExists().forPath("/nakadi/feature_toggle/" + feature);
+            final Stat stat = zkHolder.get().checkExists().forPath(PREFIX + feature.getId());
             return stat != null;
         } catch (Exception e) {
-            LOG.warn("Error occurred when checking if feature '" + feature + "' is toggled", e);
+            LOG.warn("Error occurred when checking if feature '" + feature.getId() + "' is toggled", e);
             return false;
         }
     }
+
+    public static class Feature {
```

Comment:
It will grow bigger and bigger that will make FeatureToggleService less readable.
Also it looks like a class that can be used by others that's why it is not only for internal usage of FeatureToggleService.